### ISSUE TITLE
K-fold split?

### ISSUE THREAD

**Issue by ammiiamm** [MrDupin/Machine-Learning#2]
Hello sir,

I found your blog post about k-fold on Keras from [this link](https://mrdupin.github.io/keras-k-fold/) so that it led me to this GitHub repo. I tried adapting my code with your k-fold code below, and I found that the number of train_x is less than test_x. 

```
train_x = X[i*l:(i+1)*l]
train_y = Y[i*l:(i+1)*l]
test_x = np.concatenate([X[:i*l], X[(i+1)*l:]]);
test_y = np.concatenate([Y[:i*l], Y[(i+1)*l:]]);
```

As I google around, usually people will start holding 20% of their dataset for using it as the test set. The rest 80% will use for the k-fold validation which is a part of this program, right? 60% of the remaining dataset will be allocated for training, and the rest 20% will be allocated for validating the training set in k-fold validation. 

I know that the ratio from the above is not fixed. It can be changed upon our data. I referred to the theory because I want to show that if I run this code, I will get a log like this which seems incorrect, right? No. of Train should be greater than No. of Validate samples. Should we swap the variable declaration between train_x & test_x and train_y & test_y?

```
Train on 1496 samples, validate on 5985 samples
Epoch 1/30
1496/1496 [==============================] 
```

However, I'm also in doubt too if you have an intention to use a validation set > a training set. If this is true, would you please clarify this for me? Why do you want to use validate set > train set?

**Comment by antmarakis** [comment]
You are right, nice catch! I have made the changes [here](https://github.com/MrDupin/Machine-Learning/commit/7f25e2dcdb8972a2c5c7a81890c5e3f938f47348).

**Comment by ammiiamm** [comment]
Thank you for your confirmation :)
